### ISSUE TITLE
Various Typescript inconsistencies

### ISSUE THREAD

**Issue by GordonBlahut** [perarnborg/vuex-oidc#98]
The wiki lists an `oidcError` event: ``oidcError: (payload) => console.log(`An error occured at ${payload.context}:`, payload.error)`` and states *The event oidcError is not from oidc-client, but is implemented in vuex-oidc.*

However, the type `VuexOidcStoreListeners` has no member `oidcError`.

The payload type in `VuexOidcStoreActions.authenticateOidc: (payload: { redirectPath?: string }) => Promise<void>` is too restrictive. The actual implementation also allows a simple string argument representing the redirectPath but the type does not. The payload object type should also allow any properties in addition to redirectPath because those other properties can be useful when forwarded to the oidc client.  
  
Here's a stab at a more relaxed type: `payload?: string | { redirectPath?: string | undefined; [x: string]: any }`

`public async authenticateOidcSilent(payload: { options?: VuexOidcSigninSilentOptions | undefined }): Promise<void>` forces an argument instead of it being optional. It should instead be `public async authenticateOidcSilent(payload?: { options?: VuexOidcSigninSilentOptions | undefined }): Promise<void>`

I'm also not sure if VuexOidcStoreListeners.silentRenewError is intended to be required, as the rest of the listeners are optional.

`getOidcUser: () => void;` should be `getOidcUser: () => Promise<User>;` or maybe `getOidcUser: () => Promise<User | undefined>;`

I'll try to get a PR going in the next few days if I can review in more detail.

These actions all seem to be missing from `VuexOidcStoreActions`:
```
storeOidcUser 
signOutOidcCallback
signOutPopupOidc
signOutPopupOidcCallback 
removeUser
clearStaleState
```

**Comment by GordonBlahut** [comment]
I'm also not sure if `VuexOidcStoreListeners.silentRenewError` is intended to be required, as the rest of the listeners are optional.

**Comment by perarnborg** [comment]
Thank you @GordonBlahut for looking into this. Will fix this soon!

**Comment by larserikfinholt** [comment]
It seems like VuexOidcStoreSettings is missing 
`isAuthenticatedBy: 'access_token'|'id_token'`

**Comment by perarnborg** [comment]
Fixed as of v3.5.3
